src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1119,17 +1120,21 @@ public Solution WithProjectAnalyzerReferences(ProjectId projectId, IEnumerable<A
 
         private Solution AddDocument(DocumentState state)
         {
-            if (state == null)
+            var notificationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            using (notificationService?.Start("Remove Document"))
```

Comment:
solution crawler always batch. globaloperation pause processor (that actually run code), but queue is still alive. so, when solution crawler get resumed, it will process pending work in the queue in priority order based on current context (what is active file, what is opened file and etc)